keps/sig-scalability/0001-enhance-network-performance-benchmark-testcases.md

```diff
@@ -0,0 +1,152 @@
+---
+title: Enhancing Network Performance Benchmarking Testcases
+authors:
+  - "@hanamantagoud.v.kandagal@est.tech"
+owning-sig: sig-scalability
+participating-sigs:
+- sig-networks
+- sig-testing
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: 
+creation-date: 2020-08-24
+last-updated: 2020-08-25
+status: implemented
+---
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-goals](#non-goals)
+  - [Challenges and Open Questions [optional]](#challenges-and-open-questions-optional)
+- [Proposal](#proposal)
+  - [Test Scenarios](#test-scenarios)
+  - [Tools](#tools)
+  - [Execution Environment](#execution-environment)
+  - [Future Enhancements](#future-enhancements)
+  - [Benefits](#benefits)
+- [Graduation Criteria](#graduation-criteria)
+  - [Beta -&gt; GA](#beta---ga)
+  - [GA -&gt; Future](#ga---future)
+- [Implementation History](#implementation-history)
+- [Drawbacks [optional]](#drawbacks-optional)
+- [Alternatives [optional]](#alternatives-optional)
+<!-- /toc -->
+
+## Summary
+
+We are proposing to add more testcases for network performance benchmarking.
+
+## Motivation
+
+As Kubernetes is increasingly being adopted by various telco cloud providers for deployments, 
+it brings additional networking requirements/scenarios to be supported.  
+With wide array of network functions being containerized, the need to analyse and 
+monitor the networking performance metrics such as latency, throughput etc has arisen.
+
+We have come across 2 test frameworks which have limited testcases for network performance benchmarking.
+
+a) E2E (test\e2e\network\networking_perf.go): This framework has only 2 testcases for measuring throughput.
```

Comment:
There is also clusterloader2 which does test network performance in some way. E.g. we have probes testing NetworkLatency (data plane) - [code](https://github.com/kubernetes/perf-tests/blob/b1fe05e3f799545f79daf6e44a331b5807ec1305/clusterloader2/pkg/measurement/common/probes/probes.go). It's very limited at the moment but could be easily extended to achieve more. The advantage is that it already runs continuously at 100 and 5000 node scale

Reply:
Yes we can enhance testcases in clusterloader2 as well. Can't we integrate netperf with clusterloader2 ? I thought we could reuse the code , so with lesser effort we could achieve this.

Need your help in finalizing the test framework.